### ISSUE TITLE
Dropping a keyspace with a UDT results in an Error 

### ISSUE THREAD

**Issue by paulzhol** [scylladb/scylla#3068]
Single node setup scylla 2.0.2 , to reproduce run in cqlsh:

```
CREATE KEYSPACE IF NOT EXISTS ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};

CREATE TYPE ks.type1 (
	a int,
	b int
);

CREATE TABLE ks.table1 (
	id uuid,
	udt_field frozen<type1>,
	PRIMARY KEY (id)
);

DROP KEYSPACE IF EXISTS ks;
```

**InvalidRequest: Error from server: code=2200 [Invalid query] message="Unknown type ks.type1"**

**Comment by duarten** [comment]
Seems like a regression introduced in 6c8b5fc09d5e01782acd3ef175113935f2ce9db4 (https://github.com/scylladb/scylla/commit/6c8b5fc09d5e01782acd3ef175113935f2ce9db4#diff-94c960641daa44618de006a784f3a143R774).

**Comment by duarten** [comment]
Can't repro it with branch-2.0.

**Comment by paulzhol** [comment]
It happens everytime on the scylladb/scylla:2.0.2 docker image.

**Comment by duarten** [comment]
I repro'd it with smp=2.

**Comment by duarten** [comment]
No longer sure it's a regression. I think the code needs to be worked: when adding new types and tables, we need to add those before any column families that might use them. The converse is true when dropping types and tables.

**Comment by tzach** [comment]
> The converse is true when dropping types and tables.

This does not make sense to me. If I'm deleting a KS, I'm deleting all the tables and types in it.
The above did work for me with Apache Cassandra 2.1.19

**Comment by duarten** [comment]
@tzach it shouldn't make sense to you, that's about internals :)

**Comment by gnumoreno** [comment]
Reproduced on Scylla 2.1.2-0.20180417.87cb8a1fa using AWS AMI (i3.2xlarge)

**Comment by glommer** [comment]
@gnumoreno the fix is not in 2.1. You can check that by viewing which commit fixes the issue (it gets printed when it is closed), and then searching for it in origin/branch-2.1

**Comment by tomer-sandler** [comment]
@glommer / @duarten It got reproduced also in 2.2, see ticket #3786 which was reported today.
